html/the-xhtml-syntax/parsing-xhtml-documents/xhtml-mathml-dtd-entity-support.htm

```diff
@@ -0,0 +1,62 @@
+<!DOCTYPE html>
+<meta charset=utf-8>
+<title>HTML entities for various XHTML Doctype variants</title>
+<link rel=help href="http://w3c.github.io/html/xhtml.html#parsing-xhtml-documents">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<div id="log"></div>
+<script>
+  var parser = new DOMParser();
+  var parse = parser.parseFromString.bind(parser);
+
+  function generateTestFunction(entitystring, expectedString, doctypeMarkupString, mimeType, friendlyMime) {
+    return function () {
+      var doc = parse(doctypeMarkupString + "<html><head></head><body id='test'>"+entitystring+"</body></html>", mimeType);
+      var root = doc.getElementById('test');
+      assert_not_equals(root, null, friendlyMime + " parsing the entity reference caused a parse error;");
+      assert_true(!!root.firstChild);
+      // Next line because some browsers include the partial parsed result in the parser error returned document.
+      assert_equals(root.firstChild.nodeType, 3/*Text*/, friendlyMime + " parsing the entity reference caused a parse error;");
+      var text = root.firstChild.data;
+      for (var i = 0, len = expectedString.length; i < len; i++) {
+        assert_equals(text.charCodeAt(i),expectedString.charCodeAt(i));
+      }
+    }
+  }
+
+  function setupTests(jsonEntities, doctypeMarkupString, mimeType, friendlyMime) {
+    for (entityName in jsonEntities) {
+      if ((mimeType == "text/html") || /;$/.test(entityName)) {
+        test(generateTestFunction(entityName, jsonEntities[entityName].characters, doctypeMarkupString, mimeType, friendlyMime), friendlyMime + " parsing " + entityName);
+      }
+    }
+  }
+
+  setup(function() {}, {explicit_done: true});
+
+  var xhr = new XMLHttpRequest();
+  //xhr.open("GET", "entities.json");
+  xhr.open("GET", "https://w3c.github.io/html/entities.json");
+  xhr.onload = function () {
+    // Note: for proper XML parsing of the Doctype, Edge requires a non-empty string for url part
+    var entitiesJSON = JSON.parse(xhr.response);
+    [
+      ["application/xhtml+xml", "<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Transitional//EN\" \"foo\">", "XHTML1.0 Transitional"],
```

Comment:
I'm not gonna block on this, but: I'd like to break this up into a public and system identifier fields (because `<!doctype html` is a constant). Really ideally we should probably test this with both `foo` and the original system identifier of the DTD to ensure independence from that.

What would be nice if you have time is to change the quotes around so we aren't escaping all the time (given both JS and DOCTYPEs can use both single and double quotes here that shouldn't be too hard).